File: Server/src/main/java/com/api/server/foodb/EnzymeSynonym.java
package com.api.server.foodb;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "enzyme_synonyms")
public class EnzymeSynonym {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @Column(name = "enzyme_id", nullable = false)
    private Integer enzymeId;

    @Size(max = 255)
    @NotNull
    @Column(name = "synonym", nullable = false)
    private String synonym;

    @Size(max = 255)
    @NotNull
    @Column(name = "source", nullable = false)
    private String source;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

}

File: DotNetServer/DotNetServer/Core/Entities/Foodb/EnzymeSynonym.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Core.Entities.Foodb;

[Table("enzyme_synonyms")]
[Index("EnzymeId", Name = "index_enzyme_synonyms_on_enzyme_id")]
[Index("Synonym", Name = "index_enzyme_synonyms_on_synonym", IsUnique = true)]
public partial class EnzymeSynonym
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("enzyme_id")] public int EnzymeId { get; set; }

    [Column("synonym")] public string Synonym { get; set; } = null!;

    [Column("source")] [StringLength(255)] public string Source { get; set; } = null!;

    [Column("created_at", TypeName = "datetime")]
    public DateTime? CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime? UpdatedAt { get; set; }
}